Fix the broken package download links

milljm opened the pull request:
Closes #13722

These files are being uploaded to mooseframework.org right now. But I will still need to move them into the proper place before the actually work.

Commit:
Fix the broken package download links

Closes #13722

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/366516/) on ed126ed wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13723/site)

This comment will be updated on new commits.

Commit:
The Mac OS packages completed different day

Still incorrect! The Mac OS moose environment package was created earlier.
This was due to a `conda create` issue only on linux machines.
